code/modules/cm_aliens/XenoStructures.dm

```diff
@@ -422,14 +422,22 @@
 		if(!isSwitchingStates && state == 1)
 			Close()
 
+/obj/structure/mineral_door/resin/proc/close_blocked()
+	for(var/turf/turf in locs)
+		for(var/mob/living/living_mob in turf)
+			var/datum/component/weed_food/weed = living_mob.GetComponent(/datum/component/weed_food)
+			if(!weed?.merged)
+				return TRUE
+	return FALSE
+
 /obj/structure/mineral_door/resin/Close()
 	if(!state || !loc) return //already closed
 	//Can't close if someone is blocking it
-	for(var/turf/turf in locs)
-		if(locate(/mob/living) in turf)
-			spawn (close_delay)
-				Close()
-			return
+	if(close_blocked())
+		spawn(close_delay)
+			Close()
```

Comment:
Infinite recursion with a spawn on top is kind of ehhh, should replace this with a timer while modifying there